### PR TITLE
Creating default service items

### PR DESCRIPTION
**Pull request by garrettqmartin8**
## Description

Every MTO is going to have two default service items: "Move Management Services", and "Counseling Services". Let's create them automatically!

## Reviewer Notes

It might look a bit weird having the logic for creating default service items in `movetaskorder.FetchMoveTaskorder`, but after discussing with @chrisrcoles, we decided this was the best place to do it for now. This is because there's currently no clear point at which a move task order is created.

I also considered using [Buffalo's model callbacks](https://gobuffalo.io/en/docs/db/callbacks), but it made test setup too cumbersome across the app.

## Setup

- `make server_run`
- `make client_run`
- In your favorite database tool, open up the `mto_service_items` table and delete every record.
- Visit http://officelocal:3000/too/customer/6ac40a00-e762-4f5f-b08d-3ea72a8e4b63/move-task-order/5d4b25bb-eb04-4c03-9a81-ee0398cb779e and check back in your database tool, you'll see that two default service items have been created for that MTO. Refresh the page again, and no service items are created.

## Code Review Verification Steps

* [x] Request review from a member of a different team.
* [x] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-812) for this change

### CODE REVIEW

File: pkg/services/move_task_order/move_task_order.go [hunk 1]
```diff
@@ -77,5 +77,50 @@ func (f fetchMoveTaskOrder) FetchMoveTaskOrder(moveTaskOrderID uuid.UUID) (*mode
...
 		}
 	}
+
+	f.createDefaultServiceItems(mto)
+
 	return mto, nil
 }
+
+func (f fetchMoveTaskOrder) createDefaultServiceItems(mto *models.MoveTaskOrder) error {
+	counselingItemUUID := uuid.FromStringOrNil("9dc919da-9b66-407b-9f17-05c0f03fcb50")
```
Comment (sarboc) [inline pkg/services/move_task_order/move_task_order.go]: I'm worried about having these UUIDs hard-baked into the code like this. These are for the reServiceItemID, correct? Any chance we can do a db query to find the proper UUID and then assign it? The IDs might vary between environments, etc, so a I think a lookup would be less risky.

File: pkg/services/move_task_order/move_task_order.go [hunk 2]
```diff
@@ -77,5 +77,50 @@ func (f fetchMoveTaskOrder) FetchMoveTaskOrder(moveTaskOrderID uuid.UUID) (*mode
...
+		}
+
+		for _, id := range defaultServiceItemIDs {
+			if item.ReServiceID == id {
+				defaultServiceItems = append(defaultServiceItems, item)
+			}
+		}
+	}
+
+	if len(defaultServiceItems) == 0 {
```
Comment (sarboc) [inline pkg/services/move_task_order/move_task_order.go]: If I'm understanding this logic correctly, lines 90-102 are checking to see if our default service items already exist on this MTO. Line 104 says "if we have any default service items that need to be created, let's add them." But then it adds both, regardless of how many we don't currently have on the MTO. Wouldn't it be better to do a check for each default item and then add it if it doesn't already exist?

File: pkg/services/move_task_order/move_task_order_test.go [hunk 3]
```diff
@@ -20,4 +36,8 @@ func (suite *MoveTaskOrderServiceSuite) TestMoveTaskOrderFetcher() {
 	suite.Nil(expectedMTO.ReferenceID)
 	suite.False(expectedMTO.IsAvailableToPrime)
 	suite.False(expectedMTO.IsCanceled)
+
+	var serviceItems models.MTOServiceItems
+	suite.DB().All(&serviceItems)
+	suite.Equal(2, len(serviceItems))
```
Comment (sarboc) [inline pkg/services/move_task_order/move_task_order_test.go]: I would love to see a test where we run the fetcher on an MTO that has been fetched once before and make sure that we don't add the service items again. Maybe add an additional call here and make sure the number of service items is still 2?

File: pkg/services/move_task_order/move_task_order.go (hunk 1 above)
Comment (garrettqmartin8) [reply pkg/services/move_task_order/move_task_order.go]: I got these IDs from [this sheet](https://docs.google.com/spreadsheets/d/1c8v94Axdd_0C9Vhg6FbQr5B0a01-rwmcBlyoPZARIm8/edit#gid=0). My thought was that we could save a query, but I think you're right - the tradeoff is worth being certain we have the right service items even if we have an extra query. I'll change this.

File: pkg/services/move_task_order/move_task_order.go (hunk 2 above)
Comment (garrettqmartin8) [reply pkg/services/move_task_order/move_task_order.go]: Good call!

File: pkg/services/move_task_order/move_task_order_test.go (hunk 3 above)
Comment (garrettqmartin8) [reply pkg/services/move_task_order/move_task_order_test.go]: Agreed, will implement.

File: pkg/services/move_task_order/move_task_order.go (hunk 1 above)
Comment (sarboc) [reply pkg/services/move_task_order/move_task_order.go]: Yeah - let's not prematurely optimize for queries until we know it's becoming a bottleneck. And then there might be other ways to do that optimization, like good caching, etc.

**Comment by sarboc** [comment]
@garrettqmartin8 Ok. That workflow sounds like what we want. Cool! I think I got my ghc and prime APIs mixed up and forgot which one was for the TOOs.

**Pull request merged**